Compilation errors with gcc

I get two compilation errors when compiling with gcc (compiler flags -Werror -Wall -Wextra -std=c++20).

     M42/m42.h:124:17: error: ‘bit_width’ is not a member of ‘std’
      124 |     return std::bit_width(b) - 1;

It can be fixed by adding `#include <bit>` at the beginning of m42.h

    M42/m42.cpp:51:18: error: suggest parentheses around assignment used as truth value [-Werror=parentheses]
       51 |     } while (occ = next_subset(Masks[sq], occ));

It can be fixed by adding extra parentheses:

    } while ((occ = next_subset(Masks[sq], occ)));


Also please convert the line endings to lf instead of crlf (https://docs.github.com/en/get-started/getting-started-with-git/configuring-git-to-handle-line-endings).

Thank you for releasing M42.

--- a/m42.h
+++ b/m42.h
@@ -120,7 +120,7 @@ namespace M42 {
   inline int msb(uint64_t b)
   {
     b |= 1;
-#if __cplusplus > 201703L
+#if __cplusplus > 201703L || !defined(__APPLE__)
     return std::bit_width(b) - 1;
 #elif defined(USE_INTRIN)
 #if defined(_MSC_VER)

--- a/m42.h
+++ b/m42.h
@@ -18,6 +18,10 @@
 #include <cstdlib>
 #include <cstring>
 
+#if __cplusplus > 201703L
+#include <bit>
+#endif
+
 #ifdef USE_INTRIN
 #ifdef _MSC_VER
 #include <intrin.h>
@@ -120,7 +124,7 @@ namespace M42 {
   inline int msb(uint64_t b)
   {
     b |= 1;
-#if __cplusplus > 201703L && !defined(__APPLE__)
+#if __cplusplus > 201703L
     return std::bit_width(b) - 1;
 #elif defined(USE_INTRIN)
 #if defined(_MSC_VER)

--- a/m42.h
+++ b/m42.h
@@ -120,7 +120,7 @@ namespace M42 {
   inline int msb(uint64_t b)
   {
     b |= 1;
-#if __cplusplus > 201703L || !defined(__APPLE__)
+#if __cplusplus > 201703L && !defined(__APPLE__)
     return std::bit_width(b) - 1;
 #elif defined(USE_INTRIN)
 #if defined(_MSC_VER)

